### PR TITLE
Amesos2: Fix another code path for gapped row id case

### PR DESCRIPTION
**Pull request by ndellingwood**
Partially addresses issue #3647 reported by @vbrunini

### CODE REVIEW

**Comment by ndellingwood** [comment]
Spoke to @vbrunini , this fixed issues with SuperLU and KLU2; next step is to update the UMFPACK interface to support the non-contiguous GID parameter setting. Marking this as WIP until the UMFPACK interface update is pushed in.

**Comment by ndellingwood** [comment]
@vbrunini I pushed 2 more commits, one adds non-contiguous GID support to the Umfpack interface. I tested the GappedMtxGIDs-1proc example and it now runs with correct results using Umfpack as the solver. I also updated the GappedMtxGIDs-1proc example to rerun the solve with a new RHS but reuse the previous solve's symbolic and numeric results to add coverage for the missed case in my previous PR.

File: packages/amesos2/src/Amesos2_Umfpack_def.hpp [hunk 1]
```diff
@@ -250,6 +275,8 @@ Umfpack<Matrix,Vector>::getValidParameters_impl() const
   if( is_null(valid_params) ){
     Teuchos::RCP<Teuchos::ParameterList> pl = Teuchos::parameterList();
 
+    pl->set("IsContiguous", true, "Whether GIDs contiguous");
```
Comment (srajama1) [inline packages/amesos2/src/Amesos2_Umfpack_def.hpp]: nit pick -> "GIDs are contiguous"

**Comment by ndellingwood** [comment]
@srajama1 this PR completes the majority of the issues from #3647, could you review when you get a chance? @vbrunini confirmed the fix with SuperLU. I'm having issues building and testing with SuperLU_Dist, so if there are problems there I will take care of it in a follow up PR, as the SLU_Dist interface may require additional updates as well.

File: packages/amesos2/src/Amesos2_Umfpack_def.hpp (hunk 1 above)
Comment (ndellingwood) [reply packages/amesos2/src/Amesos2_Umfpack_def.hpp]: Good catch, looks like auto-tester already started so I'll clean this up while addressing more issues. Thanks for quick review!!

**Pull request merged**